hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -97,6 +103,11 @@ def required_args(target, args=(), kwargs=()):
     and bound methods).  args and kwargs should be as they are passed to
     builds() - that is, a tuple of values and a dict of names: values.
     """
+    # We start with a workaround for NamedTuples, which don't have nice inits
```

Comment:
It would probably be good for us to file a bug upstream about this. Or is there already one?

Reply:
TBH NamedTuple is a huge pile of hacks papering over the lack of `attrs` in the standard library, but are somehow "not really class-like" (hence dataclasses...) and I don't care enough to deal with the CPython process.  Suffice to say that there are tradeoffs about tuple-like behaviour and various performance concerns that mean nobody is volunteering to change it again!